src/lightning/pytorch/plugins/precision/precision.py

```diff
@@ -40,7 +40,7 @@ def connect(
         self, model: Module, optimizers: List[Optimizer], lr_schedulers: List[Any]
     ) -> Tuple[Module, List[Optimizer], List[Any]]:
         """Connects this plugin to the accelerator and the training process."""
-        return model, optimizers, lr_schedulers
+        return self.convert_module(model), optimizers, lr_schedulers
```

Comment:
By moving it here, it leads to the conversion running after the model has been wrapped and moved to the device in the case of FSDP. I think we shouldn't put it here, but rather call the convert_module() method directly in `Strategy.setup()` so that strategies who override setup() can put this in the right place. In the FSDPStrategy for example, the `convert_module` call would come before the `self.model = self._setup_model(self.model)` call. 

This would also align it slightly more with the sequence of operations in Fabric.

Reply:
I agree that it's best that this can be overridden in the strategy. I'll make the change

> In the FSDPStrategy for example, the convert_module call would come before the self.model = self._setup_model(self.model) call.

However, I'm not sure about this because we don't know what FSDP + BNB is meant to look like yet.